lib/fshash/bucket.go

```diff
@@ -58,9 +15,30 @@ func (m Metadata) MarshalBinary() ([]byte, error) {
 */
 type Bucket interface {
 	Record(metadata Metadata, contentHash []byte)
+	Iterator() (rootRecord RecordIterator)
+}
+
+type Record struct {
+	Metadata    Metadata `json:"m"`
+	ContentHash []byte   `json:"h"`
```

Comment:
Without your response, I'd have expected the serialization to change when I modified those annotations. IMO, just slap a modified version of your response as a comment & leave it